examples/fabric-sync/admin/DeviceSynchronization.cpp

```diff
@@ -81,58 +81,65 @@ void DeviceSynchronizer::OnAttributeData(const ConcreteDataAttributePath & path,
         return;
     }
 
+    char attribStringBuffer[kBasicInformationAttributeBufSize];
+    uint32_t attribUInt32Value;
+
     switch (path.mAttributeId)
     {
-    case Clusters::BasicInformation::Attributes::UniqueID::Id: {
-        char uniqueIdBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(uniqueIdBuffer, sizeof(uniqueIdBuffer)), "UniqueId"))
+    case Clusters::BasicInformation::Attributes::UniqueID::Id:
+        if (SuccessOrLog(data->GetString(attribStringBuffer, sizeof(attribStringBuffer)), "UniqueId"))
         {
-            mCurrentDeviceData.uniqueId = std::string(uniqueIdBuffer);
+            mCurrentDeviceData.uniqueId = std::string(attribStringBuffer);
         }
-    }
-    break;
-    case Clusters::BasicInformation::Attributes::VendorName::Id: {
-        char vendorNameBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(vendorNameBuffer, sizeof(vendorNameBuffer)), "VendorName"))
+        break;
+    case Clusters::BasicInformation::Attributes::VendorID::Id:
+        if (SuccessOrLog(data->Get(attribUInt32Value), "VendorID"))
         {
-            mCurrentDeviceData.vendorName = std::string(vendorNameBuffer);
+            mCurrentDeviceData.vendorId = static_cast<chip::VendorId>(attribUInt32Value);
         }
-    }
-    break;
-    case Clusters::BasicInformation::Attributes::ProductName::Id: {
-        char productNameBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(productNameBuffer, sizeof(productNameBuffer)), "ProductName"))
+        break;
+    case Clusters::BasicInformation::Attributes::VendorName::Id:
+        if (SuccessOrLog(data->GetString(attribStringBuffer, sizeof(attribStringBuffer)), "VendorName"))
         {
-            mCurrentDeviceData.productName = std::string(productNameBuffer);
+            mCurrentDeviceData.vendorName = std::string(attribStringBuffer);
```

Comment:
Why we make this change? using the attribStringBuffer of size kBasicInformationAttributeBufSize to read all attributes with different sizes? It does not seem right to me.

Reply:
Restored previous version. All buffers use `kBasicInformationAttributeBufSize` for size anyway...